packages/components/src/autocomplete/index.tsx

```diff
@@ -256,7 +257,10 @@ export function useAutocomplete( {
 	useEffect( () => {
 		if ( ! textContent ) {
 			if ( autocompleter ) {
-				reset();
+				// Reset the state synchronously to ensure timely DOM updates without flickering.
+				// The `flushSync` can't be called directly in the effect without scheduling.
+				// See: https://github.com/WordPress/gutenberg/pull/69562.
+				window.queueMicrotask( () => flushSync( reset ) );
```

Comment:
The `queueMicrotask` prevents the following React error.

![CleanShot 2025-03-13 at 17 48 05](https://github.com/user-attachments/assets/e01d40a1-d635-4a7e-8572-7d38d9bc4a92)